src/Data/Maybe/Strict.hs

```diff
@@ -100,19 +98,22 @@ deriving instance Typeable1 Maybe
 deriving instance Generic  (Maybe a)
 #endif
 
-#if MIN_VERSION_base(4,9,0)
 instance Semigroup a => Semigroup (Maybe a) where
   Nothing <> m       = m
   m       <> Nothing = m
   Just x1 <> Just x2 = Just (x1 Semigroup.<> x2)
-#endif
 
+#if MIN_VERSION_base(4,11,0)
+instance Semigroup a => Semigroup (Maybe a) where
+  mempty = Nothing
+#else
 instance Monoid a => Monoid (Maybe a) where
   mempty = Nothing
 
   Nothing `mappend` m       = m
```

Comment:
Since we now always have a Semigroup instance we could define `mappend` as `<>`.

Reply:
Changing the context is a breaking change, as pre base-4.11 `Monoid` doesn't imply `Semigroup`. I'm :+1: to do that change, but it will mean bumping version to `0.7.0`

Reply:
I see now that defining `mappend` as `<>` will force a `Semigroup` constraint. 

I think it's better to be conservative and go with your current change. If a user wants the new definition she can always upgrade to base-4.11.

Reply:
@meiersi @meiersi-da Do you have an opinion? Either way, I'll correct other Bas' comment and bump version & add changelog entry after your comment.